generator/templates/client/response.gotmpl

```diff
@@ -35,6 +35,11 @@ func ({{ .ReceiverName }} *{{ pascalize .Name }}) Error() string {
 	return fmt.Sprintf("[{{ upper .Method }} {{ .Path }}][%d] {{ if .Name }}{{ .Name }} {{ else }}unknown error {{ end }}{{ if .Schema }} %+v{{ end }}", {{ if eq .Code -1 }}{{ .ReceiverName }}._statusCode{{ else }}{{ .Code }}{{ end }}{{ if .Schema }}, o.Payload{{ end }})
 }
 
+{{ if .Schema }}
+func ({{ .ReceiverName }} *{{ pascalize .Name }}) GetPayload() {{ if and (not .Schema.IsBaseType) (not .Schema.IsInterface) .Schema.IsComplexObject (not .Schema.IsStream) }}*{{ end }}{{ if (not .Schema.IsStream) }}{{ .Schema.GoType }}{{ else }}io.Writer{{end}} {
```

Comment:
But is that signature not defeating the purpose of having a simple common interface since we can have different signature?
Why are we allowing the possibility to generate a wrong `*io.Writer` here?
Also wouldn't it be some `ReadCloser` instead?

Reply:
It uses 1:1 the code from [here](https://github.com/go-swagger/go-swagger/pull/2019/files/e623aab4b143d7a2f1d29d66a0da51b58b1b95df#diff-6b20cc8a4514b56c8fc0488c02ece5f0R23)

Reply:
> But is that signature not defeating the purpose of having a simple common interface since we can have different signature?

Not really, most error messages share the schema and if they don't then they should have different signatures. It follows the same logic basic fetching is using.

Alternatively we could expose the response body but then we have to de/encode it again.

Reply:
ok I got your point. So we have many GetPayload() and these are going to be really useful only when the share the schema. So I am okay with that approach.
The io.Writer was probably wrong in the first place. This for the case when the response is `type: binary`

Reply:
The io.writer is not wrong, it's the sync when you consume a stream you want to write it  to somewhere